ipn/ipnlocal/peerapi.go

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) 2021 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package ipnlocal
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"hash/crc32"
+	"html"
+	"io"
+	"net"
+	"net/http"
+	"strconv"
+
+	"inet.af/netaddr"
+	"tailscale.com/tailcfg"
+)
+
+var initListenConfig func(*net.ListenConfig, netaddr.IP) error
+
+func peerAPIListen(ip netaddr.IP) (ln net.Listener, err error) {
+	var lc net.ListenConfig
+	if initListenConfig != nil {
+		// On iOS/macOS, this sets the lc.Control hook to
+		// setsockopt the interface index to bind to, to get
+		// out of the network sandbox.
+		if err := initListenConfig(&lc, ip); err != nil {
+			return nil, err
+		}
+	}
+	for try := uint8(0); try < 5; try++ {
```

Comment:
Maybe a one-liner comment explaining what you're trying to achieve with the port?

Reply:
Many liner. Done.